### PR TITLE
Deprecate ContentCodec API with offest & length

### PR DESCRIPTION
**Pull request by tkountis**
Motivation:
Deprecate the `ContentCodec` APIs that receive offset & length. User's can slice if needed.

Modifications:
Deprecate API & relevant changes to accommodate the change.

Result:
Less API surface.

### CODE REVIEW

**Comment by tkountis** [comment]
Build failure due https://github.com/apple/servicetalk/issues/1323

File: servicetalk-encoding-netty/src/main/java/io/servicetalk/encoding/netty/NettyChannelContentCodec.java [hunk 1]
```diff
@@ -177,6 +182,11 @@ public void onComplete() {
                 });
     }
 
+    @Override
+    public Buffer decode(final Buffer src, final BufferAllocator allocator) {
+        return decode(src, 0, src.readableBytes(), allocator);
```
Comment (idelpivnitskiy) [inline servicetalk-encoding-netty/src/main/java/io/servicetalk/encoding/netty/NettyChannelContentCodec.java:187]: Keep the `AbstractZipContentCodec` as it's right now in this PR, but for `NettyChannelContentCodec` consider doing the opposite: implement `decoder(Buffer, BufferAllocator)` and do the slice in this overload.

File: servicetalk-encoding-netty/src/test/java/io/servicetalk/encoding/netty/NettyChannelContentCodecTest.java [hunk 2]
```diff
@@ -76,41 +76,31 @@ public NettyChannelContentCodecTest(final ContentCodec codec) {
...
     }
 
     @Test(expected = CodecEncodingException.class)
     public void testEncodeWithOffsetAndZeroLength() {
-        testEncode(DEFAULT_ALLOCATOR, 10, 0);
-    }
-
-    @Test(expected = CodecEncodingException.class)
-    public void testEncodeWithOffsetAndOverflowLength() {
-        testEncode(DEFAULT_ALLOCATOR, 1023, 0);
```
Comment (idelpivnitskiy) [inline servicetalk-encoding-netty/src/test/java/io/servicetalk/encoding/netty/NettyChannelContentCodecTest.java:94]: Not necessary for this PR (I do not expect anyone uses codec api directly), but for future cases when you deprecate API, consider removing tests at the time you remove the method. It helps to make sure the old API still works as expected before we remove it.

**Review by idelpivnitskiy — APPROVED** [review]
One comment for `NettyChannelContentCodec` impl then LGTM

**Pull request merged**